connection.go

```diff
@@ -865,7 +743,7 @@ func (conn *Connection) notify(kind ConnEventKind) {
 	}
 }
 
-func (conn *Connection) writer(w *bufio.Writer, c net.Conn) {
+func (conn *Connection) writer(w writeFlusher, c Conn) {
```

Comment:
Is there any motivation to use custom interface instead of bufio one? As far as I understand, our one is broader